### DISCUSSION TITLE
Manage a mikrotik router address list

### DISCUSSION THREAD

**Discussion by Victoare** [DigitalRuby/IPBan#177] [Ideas]
Maybe a wild idea and i can't find anything against it (nor any mention of this), so here it is:
I have a mikrotik router and some ports natted to a server inside my lan.
In that server I used IPBan to block failed login attempts to given services.
It works marvellously, and i thought how about this blocking list would be applied not just my server but to my router as well, so i can add a preroute rule and block those addresses right in the entry point?

**Comment by jjxtra** [comment]
Should be no problem if your router supports an api. Does it?

**Reply by Victoare** [reply]
It seems that it has rest api: https://help.mikrotik.com/docs/display/ROS/REST+API

**Reply by jjxtra** [reply]
Awesome. You can use `ProcessToRunOnBan` and `ProcessToRunOnUnban` to run a curl command to send data to router.

**Reply by Victoare** [reply]
Brilliant. Thanks.
If i am managed to make this work, could i provide some details for those might interested?

Also i am double checked, the rest api is right now on a beta channel, so it may take time. 
The currently available api a bit complicated compared to a curl command.

**Reply by jjxtra** [reply]
Would love to hear about your solution once you have it. You can also run bash script or powershell script or even an executable instead of curl command for more complex scenario.
